app/components/homepage-components/TestimonialCardButtons.tsx

```diff
@@ -0,0 +1,74 @@
+"use client";
+
+import Image from "next/image";
+import { useEffect, useState } from "react";
+import { useTheme } from "next-themes";
+import { motion } from "framer-motion";
+
+import {
+  arrowLeft,
+  arrowLeftDark,
+  arrowRight,
+  arrowRightDark,
+} from "@/public/svg-icons";
+
+interface ButtonsProps {
+  additionalClassesLeft: string;
+  additionalClassesRight: string;
+  previous: () => void;
+  next: () => void;
+}
+
+const TestimonialCardButtons = ({
+  additionalClassesLeft,
+  additionalClassesRight,
+  previous,
+  next,
+}: ButtonsProps) => {
+  const [leftIcon, setLeftIcon] = useState(arrowLeft);
+  const [rightIcon, setRightIcon] = useState(arrowRight);
+  const { theme } = useTheme();
+
+  useEffect(() => {
+    if (theme === "light") {
+      setLeftIcon(arrowLeft);
+      setRightIcon(arrowRight);
+    } else {
+      setLeftIcon(arrowLeftDark);
+      setRightIcon(arrowRightDark);
+    }
+  }, [theme]);
```

Comment:
Please double-check, but I believe since the icons are being changed directly based on the theme, we can set the icons inline rather than using a state, reducing re-renders.

```
const leftIcon = theme === 'light' ? arrowLeft : arrowLeftDark;
const rightIcon = theme === 'light' ? arrowRight : arrowRightDark;
```

Reply:
I've changed this now to have custom left and right button components. I had the svgs in the code so I extracted the HTML and just used the dark mode on the stroke colour in the classname. Got rid of all the use effect and other imports as well as 4 svgs